canopen/objectdictionary/eds.py

```diff
@@ -86,8 +86,11 @@ def import_eds(source, node_id):
 
     if eds.has_section("DeviceComissioning"):
         od.bitrate = int(eds.get("DeviceComissioning", "Baudrate")) * 1000
-        od.node_id = int(eds.get("DeviceComissioning", "NodeID"), 0)
-        node_id = node_id or od.node_id
+
+        if node_id is None:
+            if val := eds.get("DeviceComissioning", "NodeID", fallback=None):
+                node_id = int(val, base=0)
```

Comment:
Great! Couldn't have written it more beautifully.  I still need to get used to the walrus operator, though I know and love it from Go (in slightly different form).

Do the same for bitrate?

Reply:
> Do the same for bitrate?

Sure; preferably in a separate PR.

Love Go, BTW! Too bad I don't get to use it often. OT: there are some interesting building concurrency blocks in the pipeline for future Python releases[^1][^2].

[^1]: https://peps.python.org/pep-0734/
[^2]: https://pyfound.blogspot.com/2024/06/python-language-summit-2024-lightning-talks.html (see Yuri's lightning talk; super interesting!)